File: my-workout-api/src/main/java/com/avernet/myworkoutapi/workout/WorkoutGroupedHistories.java
package com.avernet.myworkoutapi.workout;

import com.avernet.myworkoutapi.history.HistoryGroup;
import com.avernet.myworkoutapi.musclegroup.MuscleGroup;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Builder
public class WorkoutGroupedHistories {
    WorkoutSummary workoutSummary;
    List<HistoryGroup> historyGroups;
    MuscleGroup muscleGroup;
}

File: my-workout-app/src/interfaces/WorkoutGroupedHistories.ts
import { MuscleGroup } from './MuscleGroup';
import { HistoryGroup } from './HistoryGroup';
import { WorkoutSummary } from "./WorkoutSummary";

export class WorkoutGroupedHistories {
    workoutSummary?: WorkoutSummary;
    historyGroups?: HistoryGroup[];
    muscleGroup?: MuscleGroup;
}
